crates/autopilot/src/decoded_settlement.rs

```diff
@@ -404,23 +350,24 @@ impl DecodedSettlement {
             }
         };
 
-        // converts the order's `solver_fee` which is denominated in `sell_token` to the
-        // native token.
-        tracing::trace!(?solver_fee, "fee before conversion to native token");
-        let fee = external_prices
-            .try_get_native_amount(*sell_token, u256_to_big_rational(&solver_fee))?;
-        tracing::trace!(?fee, "fee after conversion to native token");
+        // converts the fee which is denominated in `sell_token` to the native token.
+        tracing::trace!(?fee, "fee before conversion to native token");
+        let native =
+            external_prices.try_get_native_amount(*sell_token, u256_to_big_rational(&fee))?;
+        tracing::trace!(?native, "fee after conversion to native token");
 
         Some(Fees {
-            order: order.order_uid,
-            sell: solver_fee,
-            native: big_rational_to_u256(&fee).ok()?,
+            order,
+            sell: fee,
+            native: big_rational_to_u256(&native).ok()?,
         })
     }
 }
 
-/// Computed executed fees for an order with solver-computed fees. These are
-/// computed based on-chain settlement data.
+/// These are computed based on-chain settlement data.
```

Comment:
This isn't yet fully true, it it? We still pass down the "solver_fee" if we find it in the database (and don't read it onchain)

Reply:
True. I removed the comment as misleading.
My plan for decoded_settlement.rs is to be just a file that decodes the settlement and that's it. All other functionality (surplus calculation, fees calculation) will be extracted into separate file in `domain` so it can be shared and used by RunLoop for ranking by surplus.